src/providers/CartInfoProvider.js

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useState } from 'react'
+
+export const CartInfoContext = createContext()
+
+
+// all info relating to the cart
+export const CartInfoProvider = (props) => {
+
+    const confirmedCartOrders = JSON.parse(localStorage.getItem("confirmed orders")) === null ? [{
```

Comment:
This is unreadable. I don't understand the logic of what you were trying to make here.
Please break this functionality down into multiple simple steps.
Once you do that, please move it out of the provider. It is not the provider's job to take care of local storage.
It probably should be done in some to level component like `App`